File: WarmUp_Days/warmup-day-1-basics/factorial/solutions/factorial.cpp
// Problem: Factorial of a Number
// Difficulty: Easy

#include <iostream>
using namespace std;

long long factorial(int n) {
    long long res = 1;
    for (int i = 2; i <= n; ++i) res *= i;
    return res;
}

int main() {
    cout << factorial(5) << "\n";
    return 0;
}

File: WarmUp_Days/warmup-day-1-basics/factorial/solutions/factorial.py
"""
Problem: Factorial of a Number
Difficulty: Easy
"""

"""This is math.factorial wrapper implementation."""
# import math

# def factorial(n: int) -> int:
#     # simple wrapper around math.factorial for now
#     return math.factorial(n)

def factorial(n):
    res = 1
    for i in range(2, n + 1):
        res *= i
    return res

if __name__ == "__main__":
    print(factorial(5))
